src/utils/messaging-outbox/src/entities/outbox_message.rs

```diff
@@ -13,12 +13,36 @@ use crate::OutboxMessageID;
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+// Version of outbox message which will be bumped only if the structure contains
+// breaking changes in the message structure
+pub const OUTBOX_MESSAGE_VERSION: i32 = 1;
```

Comment:
Q: If we don't allow the `OutboxMessage` user to select a version, how can we change the version for a particular message (`.content_json`) if it becomes different?

Reply:
It will not touch users. 
We should cover next scenario.
During service startup we are reading all messages from db, but if we made breaking changes we should identify that service should not work with outdated messages. So we will bump it only with `OutboxMessage` breaking changes In RP

Reply:
Interesting approach, I'll wait for comments from other teammates

Reply:
I think it will eventually become a responsibility of the consumer to determine whether they can or cannot consume a specific version of event.

E.g. it's a common practice for a service to emit V1 and V2 of the same domain event during a transitional period and then stop producing V1 once all consumers have upgraded.

I'm not saying we should spend time solving the upgradeability problem now, but we should consider better where versions really belong so that at least we didn't have to migrate from one versioning scheme to another and create meta-versions :)

Reply:
I agree that the version of events is a property of a particular producer.
We cannot have a single global version for all kinds of events, as this is a version of the payload, not the main `OutboxMessage` structure.

For now it should be easy, just declare separate constant near each message type definition, and use it in emitting code.

Reply:
If we could somehow extract the version from message class metadata, that would also simplify the emission. We already have a `Message` trait:

```
pub trait Message: Clone + Serialize + for<'a> Deserialize<'a> + Send + Sync {}
```

Perhaps we could demand returning a version number here.

Reply:
Do individual messages currently have version number in them when serialized into JSON?

Do you think per-message versioning would make sense or you think per-producer is better?

Personally I think per-message versions work much better for gradual deprecation. If we version on producer (service) level, then to deprecate a single message type we'd need to emit ALL messages as both `v1` and `v2` for some time, i.e. increase `2x` the message flow of the service.

And if we're doing per-message versioning - should we still surface the version into a column or can we rely on JSON access capabilities of Postgres to read version from serialized data?